src/OrchardCore/OrchardCore.Abstractions/Extensions/Features/ITypeFeatureProvider.cs

```diff
@@ -9,7 +10,8 @@ namespace OrchardCore.Environment.Extensions
     /// </summary>
     public interface ITypeFeatureProvider
     {
-        IFeatureInfo GetFeatureForDependency(Type dependency);
+        IEnumerable<IFeatureInfo> GetFeaturesForDependency(Type dependency);
```

Comment:
Yes, that is a valid idea. But maybe we should review the responsibilities of `ITypeFeatureProvider` and `IExtensionManager` completely, there is some overlapping.

Reply:
I added `GetExtensionForDependency`, because that makes sense as a replacement for some calls to `GetFeatureForDependency`. I also added the original `GetFeatureForDependency` back, for cases where only a single feature makes sense. This makes using the `ITypeFeatureProvider` easier.

Reply:
Great!